05.wc/lib/wc.rb

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+def wc_output
+  opt = OptionParser.new
+  wc_options = { lines: false, words: false, bytes: false }
+  opt.on('-l') { |v| wc_options[:lines] = v }
+  opt.on('-w') { |v| wc_options[:words] = v }
+  opt.on('-c') { |v| wc_options[:bytes] = v }
+  opt.parse!(ARGV)
+
+  counts_by_file = read_files
+  body = format_counts(counts_by_file, wc_options)
+  total = format_total(counts_by_file, wc_options) if counts_by_file.size > 1
+  [body, total].join
+end
+
+def read_files
+  counts = []
+  l = 0
+  w = 0
+  c = 0
+  ARGF.each do |line|
+    l += 1
+    w += line.split.size
+    c += line.bytesize
+    if ARGF.file.eof?
+      counts << { lines: l, words: w, bytes: c, path: $stdin.tty? ? ARGF.file.path : nil }
```

Comment:
```suggestion
      path = $stdin.tty? ? ARGF.file.path : ''
      counts << { lines: l, words: w, bytes: c, path: path }
```

ハッシュリテラルの中に条件分岐があると読みづらいので、いったん変数に入れましょう。
あと、文字列とnilが混在するとデータ型が異なって予期せぬ不具合の原因になったりするので、空文字列にする方がいいんじゃないかなーと思います。

Reply:
データ型を統一するように修正しました。 76bf56f91b806897543e0a23af992ee2dc921623